runtime/onert/core/src/compiler/ExecutorFactory.cc

```diff
@@ -561,8 +565,23 @@ exec::IExecutor *ExecutorFactory::createTrainableExecutor(
 {
   auto &graph = lowered_graph->graph();
 
-  backend::BackendContexts backend_contexts =
-    createBackendContexts(*lowered_graph, options.executor == "Linear");
+  backend::BackendContexts backend_contexts = createBackendContexts(*lowered_graph, true);
+
+  // Replace BackendContext with TrainbleBackendContext
+  // TODO Create context only once instead of replacing
+  for (auto &&pair : backend_contexts)
+  {
+    const auto backend = pair.first;
+    if (!backend->config()->supportTraining())
```

Comment:
Why do you check `supportTraining()` function?
I assume it is a training mode that `createTrainableExecutor` is called. So we don't need to check training mode here.
If it is not a training mode, this function should not be called.
What kinds of cases do you consider?

Reply:
There is no code that checks the training mode yet in the code implemented in this draft. This is just defensive programming. This will probably be removed after introducing training mode.

Reply:
`BackendContexts` may contain backends such as the builtin backend that do not support training. So we need a way to check whether a backend in `BackendContexts` supports training.

Reply:
But this function is `createTrainableExecutor`. So it will be called when the session is trainable.
Let's talk about it next week. :)

Reply:
> But this function is createTrainableExecutor. So it will be called when the session is trainable.

The local variable of this function, `backend_contexts`, has `buildin` backend. `builtin` backend is a backend that one core has and is now appended by `ManualScheduler`.
I think it's good to keep the `builtin` backend in training as well.

> Let's talk about it next week. :)

Sure :)